doc/modules/grid_search.rst

```diff
@@ -167,6 +171,358 @@ variable that is log-uniformly distributed between ``1e0`` and ``1e3``::
+number of resources per candidate is multiplied by ``ratio`` and the number
+of candidates is divided by the same ratio. Along with ``resource`` and
+``min_resources``, ``ratio`` is the most important parameter to control the
+search in our implementation. ``ratio`` effectively controls the number of
+iterations in :class:`HalvingGridSearchCV` and the number of candidates (if
+'auto') and iterations in :class:`HalvingRandomSearchCV`.
+``aggressive_elimination=True`` can also be used if the number of available
+resources is small. More control is available through tuning the
+``min_resources`` parameter. Each parameter and their interactions are
+described in more details below.
+
+These estimators are still **experimental**: their predictions
+and their API might change without any deprecation cycle. To use them, you
+need to explicitly import ``enable_successive_halving``::
+
+  >>> # explicitly require this experimental feature
+  >>> from sklearn.experimental import enable_successive_halving # noqa
+  >>> # now you can import normally from model_selection
+  >>> from sklearn.model_selection import HalvingGridSearchCV
+
+.. topic:: Examples:
+
+    * :ref:`sphx_glr_auto_examples_model_selection_plot_successive_halving_heatmap.py`
+    * :ref:`sphx_glr_auto_examples_model_selection_plot_successive_halving_iterations.py`
+
+Choosing ``min_resources`` and the number of candidates
+-------------------------------------------------------
+
+Beside ``ratio``, the two main parameters that influence the behaviour of a
+successive halving search are the ``min_resources`` parameter, and the
+number of candidates (or parameter combinations) that are evaluated.
+``min_resources`` is the amount of resources allocated at the first
+iteration for each candidate. The number of candidates is specified directly
+in :class:`HalvingRandomSearchCV`, and is determined from the ``param_grid``
+parameter of :class:`HalvingGridSearchCV`.
+
+Consider a case where the resource is the number of samples, and where we
+have 1000 samples. In theory, with ``min_resources=10`` and ``ratio=2``, we
+are able to run **at most** 7 iterations with the following number of
+samples: ``[10, 20, 40, 80, 160, 320, 640]``.
+
+But depending on the number of candidates, we might run less than 7
+iterations: if we start with a **small** number of candidates, the last
+iteration might use less than 640 samples, which is a waste of resources. For
+example if we start with 5 candidates, we only need 2 iterations: 5
+candidates for the first iteration, then `5 // 2 = 2` candidates at the
+second iteration, after which we know which candidate performs the best (so
+we don't need a third one). We would only be using at most 20 samples which
+is a waste since we have 1000 samples at our disposal.
+On the other hand, if we start with a **high** number of candidates, we might
+end up with a lot of candidates at the last iteration, which is not always
```

Comment:
why is it not ideal? "which means many candidates will run with the full resources, basically reducing the procedure to standard search"?